static/application.js

```diff
@@ -333,15 +333,12 @@ class haste {
 // due to the behavior of lookupTypeByExtension and lookupExtensionByType
 // Note: optimized for lookupTypeByExtension
 haste.extensionMap = {
-	rb: 'ruby', py: 'python', pl: 'perl', php: 'php', scala: 'scala', go: 'go',
-	xml: 'xml', html: 'xml', htm: 'xml', css: 'css', js: 'javascript', vbs: 'vbscript',
-	lua: 'lua', pas: 'delphi', java: 'java', cpp: 'cpp', cc: 'cpp', m: 'objectivec',
-	vala: 'vala', sql: 'sql', sm: 'smalltalk', lisp: 'lisp', ini: 'ini',
-	diff: 'diff', bash: 'bash', sh: 'bash', tex: 'tex', erl: 'erlang', hs: 'haskell',
-	md: 'markdown', txt: '', coffee: 'coffee', json: 'javascript',
-	swift: 'swift'
-};
-
+	sh: 'bash', clike: 'c-like', coffee: 'coffeescript', cs: 'csharp', dpr: 'delphi',
+	erl: 'erlang', hs: 'haskell', js: 'javascript', kt: 'kotlin', tex: 'latex',
+	lsp: 'lisp', mk: 'makefile', md: 'markdown', mm: 'objectivec', phptemp: 'php-template',
+	pl: 'perl',	txt: 'plaintext', py: 'python', pyrepl: 'python-repl', rb: 'ruby', rs: 'rust',
+	sc: 'scala', sm: 'smalltalk', ts: 'typscript', vbs: 'vbscript', html: 'xml', htm: 'xml',
+}
```

Comment:
- Looks like there are now less languages than there used to be, please re-add the missing ones that were deleted, like `json` or `ini`.
- Line 339: Please change `'plaintext'` to `'txt'` (or just revert to `''`, either will do).
- Re-add missing semicolon at the end.

Reply:
This list only exists to make an alias for the language. For things like json where the file extension already is `.json` there doesn't need to be any alias. past.com/sdafsda.haskell works just as fine as past.com/sdafsda.hs

The extension for plaintext is already set to txt. The dictionary is formatted like `alias:hljs keyword`

Reply:
I get the confusion though. When he had stuff in there like `java: 'java'` it made me think this is the list but on my instance running with these changes java works without an entry in this dictionary. (https://p.twil.cx/liz.java)

Reply:
Sorry for the late reply, I checked code further and you were right. When I was testing this change it seemed like haste was wrongly adding extensions to code in other languages, but it seems that was some kind of bug or mistake on my end because I was tired. I tested that now again and everything seems to work as intended :)